### PR TITLE
dprint: init at 0.15.0

### PR DESCRIPTION
**Pull request by khrj**
###### Motivation for this change

Adds new package for https://github.com/dprint/dprint

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/dprint/default.nix [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchCrate {
+    inherit pname version;
+    sha256 = "sha256-1DUGp+HiiY03fyZ+b8hNUBIfuQV5Z/gEcOxc/vG3YiA=";
+  };
+
+  cargoSha256 = "sha256-twFXA8A+vP1n6IFJO78fKNs+FC2ui46rj1JmJ/eq3wc=";
+
+  # Tests fail because they expect a test WASM plugin. Tests already run for
+  # every commit upstream on GitHub Actions
```
Comment (SuperSandro2000) [inline pkgs/development/tools/dprint/default.nix:15]: It is still nice to know if the program works in the nix environment, too.

Comment (KhushrajRathod) [reply pkgs/development/tools/dprint/default.nix:15]: Hi, I've tried fixing this, but the tests [depend on the structure of the target directory too ](https://github.com/dprint/dprint/blob/d19785a19737058659f794bbcdbe4771d8dfdabf/crates/dprint/src/cli/run_cli.rs#L2436)

**Comment by KhushrajRathod** [comment]
Hey @SuperSandro2000, this is my first time contributing to nixpkgs! I don't see `dprint` at https://search.nixos.org/packages?channel=unstable&query=dprint yet, so am I doing something wrong or is it just taking a while? Thanks

**Comment by tejasag** [comment]
@KhushrajRathod it is visible to me :sweat_smile: try clearing your cache

**Comment by KhushrajRathod** [comment]
Oh it turns out it just took a while (i'm pretty sure there's no cache for that locally since its a dynamic page) - I see it too now, thanks!

**Pull request merged**